### PR TITLE
feat: Doc updates, Issue and PR templates are added

### PR DESCRIPTION
**Pull request by vara-bonthu**
# What does this PR do

**Changes Included:**

**README Update:** The main README.md now features the Kubeflow header and includes a new link to our updated Slack channel, facilitating better community engagement.

**PR Template Addition:** Added a new pull request template to the .github folder to streamline future contributions.

**Workflow Update:** Updated references to the container registry in our GitHub workflows to use ghcr.io, ensuring consistency with our current infrastructure.

**Contributing Guidelines Revision:** Removed the requirement for Google CLA from our contributing guidelines to simplify the contribution process.

**Issue Templates Creation:** Created new issue templates for Bugs, Features, and Questions to organize and prioritize incoming issues more efficiently.

### CODE REVIEW

**Comment by vara-bonthu** [comment]
One of the workflow is failed due to the bug introduced by recent PRs. Ignore this and this will be addressed in the next PR

image

File: README.md [hunk 1]
```diff
@@ -68,30 +81,13 @@ When installing using the Helm chart, you can choose to use a specific image tag
 
 Get started quickly with the Kubernetes Operator for Apache Spark using the [Quick Start Guide](docs/quick-start-guide.md).
 
-If you are running the Kubernetes Operator for Apache Spark on Google Kubernetes Engine and want to use Google Cloud Storage (GCS) and/or BigQuery for reading/writing data, also refer to the [GCP guide](docs/gcp.md).
-
```
Comment (yuchaoran2011) [inline README.md:72]: Isn't this part still relevant?

Comment (vara-bonthu) [reply README.md:72]: I noticed that the GCP documentation is still in the docs folder, so I will undo the changes that removed it.

**Review by yuchaoran2011 — APPROVED** [review]
/lgtm

**Comment by AndrewChubatiuk** [comment]
@vara-bonthu fixed it in my PR. this PR has only changes in README.md, but has no changes in README.md template. workflow validates if there are changes in README.md after `make helm-docs` execution

**Comment by ChenYi015** [comment]
@vara-bonthu Just run `make helm-docs` and commit the changes. I have updated `README.go.tmpl` in https://github.com/kubeflow/spark-operator/pull/1958. If there are any changes to the chart, like chart version, the command `make helm-docs` should be run to update the README.md file.

**Comment by vara-bonthu** [comment]
@ChenYi015 It might be beneficial to include the `pre-commit` instructions in the `contributing.md` file to assist contributors.

**Comment by ChenYi015** [comment]
> @ChenYi015 It might be beneficial to include the `pre-commit` instructions in the `contributing.md` file to assist contributors.

Agree, I will do it.

**Review by yuchaoran2011 — APPROVED** [review]
/lgtm

**Pull request merged**